### PR TITLE
docs: 0.5.0 documentation updates

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?
- adds an introduction to the aligner docs
- replaces (most) references to llama2 models with llama3 models
- makes some minor fixes and clarifications

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/user-guide/dpo.rst [hunk 1]
```diff
@@ -5,9 +5,12 @@
 Model Alignment by DPO, RPO, and IPO
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:7]: Line 16/19 revise sentences and use "In" and "the"

In full-parameter DPO, there exists an actor and a reference model. The actor is initialized with the reference model and is fully trainable. The reference model is frozen and used to calculate logprobs for KL-penalty loss (see the `DPO paper <https://arxiv.org/pdf/2305.18290.pdf>`__).

File: docs/user-guide/draftp.rst [hunk 2]
```diff
@@ -5,6 +5,9 @@
 Fine-tuning Stable Diffusion with DRaFT+
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
+.. note::
+   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
+
 In this tutorial, we will go through the step-by-step guide for fine-tuning Stable Diffusion model using DRaFT+ algorithm by NVIDIA.
```
Comment (jgerh) [inline docs/user-guide/draftp.rst:11]: Add definitive article "the" for clarity.

In this tutorial, we will go through the step-by-step guide for fine-tuning the Stable Diffusion model using DRaFT+ algorithm by NVIDIA.

File: docs/user-guide/draftp.rst [hunk 3]
```diff
@@ -5,6 +5,9 @@
 Fine-tuning Stable Diffusion with DRaFT+
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
+.. note::
+   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
+
 In this tutorial, we will go through the step-by-step guide for fine-tuning Stable Diffusion model using DRaFT+ algorithm by NVIDIA. 
 DRaFT+ is an improvement over the `DRaFT <https://arxiv.org/pdf/2309.17400.pdf>`__ algorithm by alleviating the mode collapse and improving diversity through regularization.
```
Comment (jgerh) [inline docs/user-guide/draftp.rst:12]: suggested revision.

DRaFT+ enhances the DRaFT <https://arxiv.org/pdf/2309.17400.pdf>__ algorithm by mitigating mode collapse and improving diversity through regularization.

File: docs/user-guide/modelalignment.rsts [hunk 4]
```diff
@@ -1,2 +1,37 @@
+
+.. _model-aligner-intro:
+
 Model Alignment
 !!!!!!!!!!!!!!!
+
+Introduction
+############
+
+NeMo-Aligner is a scalable toolkit for efficient model alignment. The toolkit has support for state of the art model alignment algorithms such as SteerLM, DPO and Reinforcement Learning from Human Feedback (RLHF). These algorithms enable users to align language models to be more safe, harmless and helpful. Users can do end-to-end model alignment on a wide range of model sizes and take advantage of all the parallelism techniques to ensure their model alignment is done in a performant and resource efficient manner. For more technical details, please refer to our [paper](https://arxiv.org/abs/2405.01481).
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:10]: fix punctuation, define first use of acronym, and suggested revision for clarity.

NeMo-Aligner is a scalable toolkit for efficient model alignment. The toolkit has support for state-of-the-art model alignment algorithms such as SteerLM, Direct Preference Optimization (DPO), and Reinforcement Learning from Human Feedback (RLHF). These algorithms enable users to align language models to be more safe, harmless, and helpful. Users can perform end-to-end model alignment on a wide range of model sizes and take advantage of all the parallelism techniques to ensure their model alignment is done in a performant and resource-efficient manner. For more technical details, please refer to our [paper](https://arxiv.org/abs/2405.01481).

File: docs/user-guide/modelalignment.rsts [hunk 5]
```diff
@@ -1,2 +1,37 @@
...
+
+NeMo-Aligner toolkit is built using the [NeMo Toolkit](https://github.com/NVIDIA/NeMo) which allows for scaling training up to 1000s of GPUs using tensor, data and pipeline parallelism for all components of alignment. All of our checkpoints are cross compatible with the NeMo ecosystem; allowing for inference deployment and further customization.
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:12]: fix punctuation and revise for clarity.

The NeMo-Aligner toolkit is built using the [NeMo Toolkit](https://github.com/NVIDIA/NeMo) which allows for scaling training up to 1000s of GPUs using tensor, data, and pipeline parallelism for all components of alignment. All of our checkpoints are cross-compatible with the NeMo ecosystem, allowing for inference deployment and further customization.

File: docs/user-guide/modelalignment.rsts [hunk 6]
```diff
@@ -1,2 +1,37 @@
...
+
+The toolkit is currently in its early stages, and we are committed to improving the toolkit to make it easier for developers to pick and choose different alignment algorithms to build safe, helpful and reliable models.
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:14]: fix punctuation and trim run-on sentence.

The toolkit is currently in its early stages. We are committed to improving the toolkit to make it easier for developers to pick and choose different alignment algorithms to build safe, helpful, and reliable models.

File: docs/user-guide/modelalignment.rsts [hunk 7]
```diff
@@ -1,2 +1,37 @@
...
+
+
+Getting Started
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:17]: suggest change to imperative verb.

Get Started

File: docs/user-guide/modelalignment.rsts [hunk 8]
```diff
@@ -1,2 +1,37 @@
...
+
+To use a pre-built container, run the following code:
+
+    .. code-block:: bash
+
+        docker run -it --gpus=all  --shm-size=8g --workdir /opt/NeMo-Aligner nvcr.io/nvidia/nemo:24.09
+
+    Please use the latest tag in the form yy.mm.(patch).
+
+Notes and Tips
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:32]: This isn't really or a note or a tip; it's an instruction. suggested revision to imperative verb.

Access Hugging Face Models

File: docs/user-guide/modelalignment.rsts [hunk 9]
```diff
@@ -1,2 +1,37 @@
...
+##############
+
+- Some of the subsequent tutorials require accessing gated Hugging Face models. For details on how to access these models, refer to the ``this document <https://docs.nvidia.com/nemo-framework/user-guide//latest/generaltips.html#working-with-hugging-face-models>``__.
```
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:35]: remove "-" character. Also, it is note clear what the "subsequent" tutorials are in this context. I don't see any other tutorials listed in this topic.

Some of the subsequent tutorials require accessing gated Hugging Face models. For details on how to access these models, refer to the ``this document <https://docs.nvidia.com/nemo-framework/user-guide//latest/generaltips.html#working-with-hugging-face-models>``__.

File: docs/user-guide/draftp.rst [hunk 10]
```diff
@@ -5,6 +5,9 @@
 Fine-tuning Stable Diffusion with DRaFT+
```
Comment (jgerh) [inline docs/user-guide/draftp.rst:5]: fix capitalization.

Fine-Tuning Stable Diffusion with DRaFT+

File: docs/user-guide/modelalignment.rsts (hunk 9 above)
Comment (jgerh) [inline docs/user-guide/modelalignment.rsts:35]: does not appear to be both "notes and tips". suggest remove heading and change to either a note or a  tip.

.. note::

Some of the subsequent tutorials require accessing gated Hugging Face models. For details on how to access these models, refer to ``this document <https://docs.nvidia.com/nemo-framework/user-guide//latest/generaltips.html#working-with-hugging-face-models>``__.

File: docs/user-guide/rlhf.rst [hunk 11]
```diff
@@ -5,7 +5,10 @@
 Model Alignment by RLHF
 @@@@@@@@@@@@@@@@@@@@@@@
 
-For the purposes of this tutorial, we will go through the entire Reinforcement Learning from Human Feedback (RLHF) pipeline using models from the NeMo Framework. These models can include LLaMa2 or Mistral, and our scripts will function consistently across them.
+.. note::
```
Comment (jgerh) [inline docs/user-guide/rlhf.rst:8]: Line 15/18 fix JSON Lines format.

We have a script ready to use for processing the Anthropic-HH dataset into a JSON Lines format.

File: docs/user-guide/rlhf.rst [hunk 12]
```diff
@@ -5,7 +5,10 @@
...
+   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
+
```
Comment (jgerh) [inline docs/user-guide/rlhf.rst:10]: Line 15/18

change jsonlines format to "JSONL format".

We have a script ready to use for processing the Anthropic-HH dataset into a JSONL format. Run the following command on the `download_and_process.py <https://github.com/NVIDIA/NeMo-Megatron-Launcher/blob/8d4f34c9da6b3254ca316b2c43ee88b77a894529/launcher_scripts/nemo_launcher/collections/dataprep_scripts/anthropichh_dataprep/download_and_process.py#L1>`__ script for anthropic HH.

File: docs/user-guide/sft.rst [hunk 13]
```diff
@@ -7,7 +7,7 @@ Obtain a Pretrained Model
 
 The NeMo Framework supports efficient model alignment using the NeMo-Aligner codebase. All algorithms in NeMo-Aligner will work with any NeMo GPT-based model. To see a collection of scripts that convert popular models from Hugging Face to ``.nemo`` format, go `here <https://github.com/NVIDIA/NeMo/tree/main/scripts/nlp_language_modeling>`__.
 
-To get started, you need to obtain a pretrained model to align. Three models are recommended: 2B GPT, LLama2-7B, or Nemotron-340B. For demonstration purposes, the smaller 2B model will be used, but you can follow the rest of the tutorial with either model.
+To get started, you need to obtain a pretrained model to align. Three models are recommended: 2B GPT, LLama3-8B, or Nemotron-340B. For demonstration purposes, the smaller 2B model will be used, but you can follow the rest of the tutorial with any of thes three models.
```
Comment (jgerh) [inline docs/user-guide/sft.rst:10]: fix spelling.

To get started, you need to obtain a pretrained model to align. Three models are recommended: 2B GPT, LLama3-8B, or Nemotron-340B. For demonstration purposes, the smaller 2B model will be used, but you can follow the rest of the tutorial with any of these three models.

File: docs/user-guide/spin.rst [hunk 14]
```diff
@@ -7,7 +7,10 @@ Original paper: https://arxiv.org/abs/2401.01335
 
 The NeMo framework supports efficient model alignment via the NeMo Aligner codebase.
```
Comment (jgerh) [inline docs/user-guide/spin.rst:8]: fix capitalization and punctuation.

The NeMo Framework supports efficient model alignment via the NeMo-Aligner codebase.

File: docs/user-guide/spin.rst [hunk 15]
```diff
@@ -7,7 +7,10 @@ Original paper: https://arxiv.org/abs/2401.01335
...
 
-All algorithms in NeMo Aligner will work with any GPT based model that is from mcore(i.e in the config it has ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire SPIN pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  This same tutorial also works for GPT models(such as LLaMa2) of any size.
+All algorithms in NeMo Aligner will work with any GPT based model that is from mcore(i.e in the config it has ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire SPIN pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  This same tutorial also works for GPT models(such as LLaMa3) of any size.
```
Comment (jgerh) [inline docs/user-guide/spin.rst:10]: fix punctuation (NeMo-Aligner is hyphenated) and spacing.

All algorithms in NeMo-Aligner will work with any GPT-based model that is from Megatron Core (in the config, ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire SPIN pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  This same tutorial also works for GPT models (such as LLaMa3) of any size.

File: docs/user-guide/spin.rst [hunk 16]
```diff
@@ -7,7 +7,10 @@ Original paper: https://arxiv.org/abs/2401.01335
...
+
```
Comment (jgerh) [inline docs/user-guide/spin.rst:11]: Move Line 6/6 to Line 11 (or abouts) and revise as a sentence. 

The original paper on Self-Play Fine-Tuning is here: 
`https://arxiv.org/abs/2401.01335 <https://arxiv.org/abs/2401.01335>`_.

File: docs/user-guide/spin.rst [hunk 17]
```diff
@@ -7,7 +7,10 @@ Original paper: https://arxiv.org/abs/2401.01335
...
+.. note::
+   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
 
 Obtaining a pretrained model
```
Comment (jgerh) [inline docs/user-guide/spin.rst:15]: fix capitalization and change title to an imperative verb.

Obtain a Pretrained Model

File: docs/user-guide/spin.rst [hunk 18]
```diff
@@ -20,18 +23,18 @@ To start, we must first get a pretrained model to align. There are 2 models we r
 
         #. Get the 2B checkpoint via ``wget https://huggingface.co/nvidia/GPT-2B-001/resolve/main/GPT-2B-001_bf16_tp1.nemo``
```
Comment (jgerh) [inline docs/user-guide/spin.rst:24]: fix punctuation.

    #. Get the 2B checkpoint via ``wget https://huggingface.co/nvidia/GPT-2B-001/resolve/main/GPT-2B-001_bf16_tp1.nemo``.

File: docs/user-guide/spin.rst [hunk 19]
```diff
@@ -20,18 +23,18 @@ To start, we must first get a pretrained model to align. There are 2 models we r
...
         #. Extract the NeMo File to a folder with ``mkdir model_checkpoint && tar -xvf GPT-2B-001_bf16_tp1.nemo -C model_checkpoint``
```
Comment (jgerh) [inline docs/user-guide/spin.rst:25]: fix  punctuation.

    #. Extract the NeMo File to a folder with ``mkdir model_checkpoint && tar -xvf GPT-2B-001_bf16_tp1.nemo -C model_checkpoint``.

File: docs/user-guide/spin.rst [hunk 20]
```diff
@@ -20,18 +23,18 @@ To start, we must first get a pretrained model to align. There are 2 models we r
...
-        #. And then run the script to convert from old NeMo checkpoint to Megatron-Core checkpoint. The script is located `here <https://github.com/NVIDIA/NeMo/blob/86b198ff93438d454f9c7f3550bcfb7d4e59feab/scripts/nlp_language_modeling/convert_nemo_gpt_to_mcore.py>`__.
```
Comment (jgerh) [inline docs/user-guide/spin.rst:23]: revise.

#. Run the script to convert from the old NeMo checkpoint to the Megatron-Core checkpoint. The script is located `here <https://github.com/NVIDIA/NeMo/blob/86b198ff93438d454f9c7f3550bcfb7d4e59feab/scripts/nlp_language_modeling/convert_nemo_gpt_to_mcore.py>`__.

File: docs/user-guide/spin.rst (hunk 16 above)
Comment (ashors1) [reply docs/user-guide/spin.rst:11]: Reworded the sentence a bit. Please let me know if you have any comments

File: docs/user-guide/spin.rst [hunk 21]
```diff
@@ -20,18 +23,18 @@ To start, we must first get a pretrained model to align. There are 2 models we r
...
                   --out-file ./mcore_gpt.nemo
 
-    .. tab-item:: LLaMa2 7B
+    .. tab-item:: LLaMa3 8B
         :sync: key2
 
-        #. Download the `Llama 2 7B LLM model and tokenizer <https://huggingface.co/meta-llama/Llama-2-7b>`__ into the models folder.
-        #. Convert the LLaMa2 LLM into ``.nemo`` format
+        #. Download the `Llama 3 8B LLM model and tokenizer <https://huggingface.co/meta-llama/Meta-Llama-3-8B>`__ into the models folder.
+        #. Convert the LLaMa3 LLM into ``.nemo`` format
```
Comment (jgerh) [inline docs/user-guide/spin.rst:37]: fix punctuation.

   #. Convert the LLaMa3 LLM into ``.nemo`` format.

File: docs/user-guide/rs.rst [hunk 22]
```diff
@@ -5,7 +5,10 @@
 Model Alignment by Rejection Sampling
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
-In this tutorial, we will guide you through the process of aligning a NeMo Framework model using rejection sampling. This method can be applied to various models, including LLaMa2 and Mistral, with our scripts functioning consistently across different models.
+.. note::
+   Before starting this tutorial, be sure to review the :ref:`introduction <model-aligner-intro>` for tips on setting up your NeMo-Aligner environment.
+
+In this tutorial, we will guide you through the process of aligning a NeMo Framework model using rejection sampling. This method can be applied to various models, including LLaMa and Mistral, with our scripts functioning consistently across different models.
```
Comment (jgerh) [inline docs/user-guide/rs.rst:11]: Line 15/18 fix punctuation and remove definition of acronym; already defined. 

After you have fine-tuned a GPT model using SFT and trained a reward model as explained in the preceding section, you can start aligning the policy using rejection sampling.

File: docs/user-guide/dpo.rst [hunk 23]
```diff
@@ -36,18 +39,18 @@ To start, we must first get a pretrained model to align. There are two models we
...
                   --out-file ./mcore_gpt.nemo
 
-    .. tab-item:: LLaMa2 7B
+    .. tab-item:: LLaMa3 7B
         :sync: key2
 
-        #. Download the `Llama 2 7B LLM model and tokenizer <https://huggingface.co/meta-llama/Llama-2-7b>`__ into the models folder.
-        #. Convert the LLaMa2 LLM into ``.nemo`` format.
+        #. Download the `Llama 3 8B LLM model and tokenizer <https://huggingface.co/meta-llama/Meta-Llama-3-8B>`__ into the models folder.
+        #. Convert the LLaMa3 LLM into ``.nemo`` format.
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:53]: Line 197/203 fix capitalization for bullets.

* global_batch_size: Generally, we have found that, all other parameters held equal, lower GBS performs worse. GBS of 256 or 512 seems to be the sweet spot for most models we trained.
* epochs: Highly sensitive to training data size. We recommend you start with 1 epoch and then add on from there. We did not see any improvements beyond 3 epochs.
* learning rate: We tested cosine annealing with a warmup of 10 steps, followed by a slow decay to a constant rate. That constant rate should be fairly low. We saw the best performance with 9e-7 and 5-e7.
* ref_policy_kl_penalty: We generally saw better performance with lower values of 0.1, 0.2, 0.5, and 1.0. Occasionally, values as high as 5.0 worked too.

File: docs/user-guide/steerlm.rst [hunk 24]
```diff
@@ -50,6 +50,8 @@ This section is a step-by-step tutorial that walks you through how to run a full
 
 4. Inference on the SteerLM model with different attribute values
```
Comment (jgerh) [inline docs/user-guide/steerlm.rst:52]: Line 9/9 fix capitalization and add acronym for terms. 

**SteerLM** is a novel approach developed by the NVIDIA NeMo Team, introduced as part of NVIDIA NeMo Alignment methods. It simplifies the customization of large language models (LLMs) and empowers users with dynamic control over model outputs by specifying desired attributes. 

Despite remarkable progress in natural language generation driven by LLMs like GPT-3, Megatron-Turing, Chinchilla, PaLM-2, Falcon, and Llama 2, these foundational models often fall short in delivering nuanced and user-aligned responses. 

The current approach for LLM improvement combines Supervised Fine-Tuning (SFT) and Reinforcement Learning from Human Feedback (RLHF), but it comes with complexities and limited user control. 

SteerLM addresses these challenges and represents a significant advancement in the field, making it easier to tailor LLMs to specific needs and preferences. This document delves into how SteerLM operates and offers guidance on training a SteerLM model.

Line 13/13 use SFT acronym previously defined.

SteerLM leverages a SFT method that empowers you to control responses during inference. It overcomes the limitations of prior alignment techniques, and consists of four key steps:

Line 32 use RLHF acronym previously defined.

RLHF and SteerLM are two methods aimed at aligning language models to human preferences. RLHF trains language models by providing positive or negative feedback on generated responses, reinforcing good behaviors. Specifically, the model is encouraged to generate more text similar to responses that receive positive feedback, and less like those with negative feedback.

Line 34 revise.

The two methods approach model alignment from different angles: RLHF reinforces desired model behaviors directly, while SteerLM steers generation based on categorical labels. Both aim to produce language model outputs that are better aligned with human values and preferences.

Line 36 revise preposition.

.. note::
   For details on SteerLM, please refer to our paper `SteerLM: Attribute Conditioned SFT as an (User-Steerable) Alternative to RLHF <https://arxiv.org/abs/2310.05344>`_.

Line 38 revise.

For details about the HelpSteer dataset, please refer to our paper `HelpSteer: Multi-attribute Helpfulness Dataset for SteerLM <https://arxiv.org/abs/2311.09528>`_.

Line 43/43 revise.

This section is a step-by-step tutorial that walks you through how to run a full SteerLM pipeline with a Llama2 70B LLM model. 

Line 51/51 Delete this list of 4 steps. It is redundant and not needed. Also, the list contains only 4 steps and the procedures that follow contain 6 steps. It is best to be consistent here. Not needed, so remove the steps here.

Line 54/56 - 55/57 remove step (this is a procedural heading that contains numbered steps), fix capitalization, and change to sub-sub heading level.


Download the Llama 2 LLM Model
	
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 56/58 change to a numbered step in this section. (Ok to use numbers like 1., 2., etc. or #. character for steps. I use numbers below as an example).

1. Download the Llama 2 70B LLM model from HF <https://huggingface.co/meta-llama/Llama-2-70b-hf> into the models folder.

Line 58/60 revise and change to a numbered step.

2. Convert the Llama 2 LLM into .nemo format:


Line 65/67 change to a numbered step, revise, and fix spelling.


3. Download and convert to .nemo format for the 13B model <https://huggingface.co/meta-llama/Llama-2-13b-hf>. This is needed for the Attribute Prediction Modeling step.

67/69 change to a numbered step.

4. Untar the .nemo file to obtain the tokenizer in NeMo format (only for the 70B model):

	
Line 79/81 - 80/82 remove step, fix capitalization, and change to sub-sub heading level.


Download and Preprocess Data for Attribute Prediction Modeling 
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 82/84 revise, fix punctuation, and change to a numbered step in this section.


1. Download and convert both datasets into a common format:

Line 90/92 revise, fix punctuation, and change to a numbered step.

2. Merge the two datasets for the train and val subset, respectively:

Line 98/100 revise, fix punctuation, and change to a numbered step.

3. Preprocess the data into regression reward model training format:


Line 111/113 - 112/114 remove step, fix capitalization, and change to sub-sub heading level.


Train the Regression Reward Model on OASST+HelpSteer Data
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 116/118 change to note format, revise, and fix punctuation.

.. note::
  Depending on the type of cluster you use, you may need to set up multi-node training in your cluster env. For details, please refer to https://lightning.ai/docs/pytorch/stable/clouds/cluster.html. 


Line 146/148 - 147/149 remove step, fix capitalization, and change to sub-sub heading level.


Generate Annotations
^^^^^^^^^^^^^^^^^^^^^
 
Line 148/150 change to a numbered step in this section.

1. To generate annotations, run the following command in the background to launch an inference server:

Line 162/164 change to a numbered step.

2. Execute the following code:

Line 179/181 - 180/182 remove step, fix capitalization, and change to sub-sub heading level.


Train the Attribute-Conditioned SFT Model
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

236/238 - 237/239 remove step, revise title (include a verb) and change to sub-sub heading level.

Run Inference
^^^^^^^^^^^^^^

Line 238/240 change to numbered step in this section.

1. To start inference, run an inference server in the background using the following command:

Line 256/258 change to a numbered step.

2. Create Python helper functions:

Line 288/290 change to a numbered step.

3. Change the values below to steer the language model:

Line 307/309 change to a numbered step.

4. Ask questions and generate responses:

File: docs/user-guide/steerlm2.rst [hunk 25]
```diff
@@ -159,6 +159,8 @@ By organizing the data in this format, the SteerLM 2.0 model can be effectively
         exp_manager.explicit_log_dir=/results/acsft_70b \
         exp_manager.checkpoint_callback_params.save_nemo_on_train_end=True 
 
+``/path/to/steerlm1/model`` is the path to the initial SteerLM model. For details on training the initial SteerLM model, refer to the :ref:`SteerLM documentation <model-aligner-steerlm>`.
+
```
Comment (jgerh) [inline docs/user-guide/steerlm2.rst:163]: Line 9/9 fix capitalization and add acronym for terms. 

**SteerLM 2.0** is a novel approach for aligning large language models (LLMs) to generate responses with desired attribute values, building upon the original `SteerLM <model-aligner-steerlm>`_ method [1]_ . While SteerLM conducts attribute-conditioned Supervised Fine-Tuning (SFT) to steer LLM outputs, SteerLM 2.0 introduces an iterative training procedure to explicitly enforce the generated responses to follow the desired attribute distribution.

Line 50/50 change to imperative verb

Prepare the Training Dataset

Line 53/53 Supervised Fine-Tuning, OK to use acronym here.

- A SFT model :math:`P(y|x)` that generates responses :math:`y` given a prompt :math:`x`

Line 109/109. add definitive article.

The following is an example of launching the training of SteerLM 2.0:

**Review by jgerh — COMMENTED** [review]
I have completed the document review of the nine files in this PR. Please review the edits and resolve the open threads as you see fit. Discussion on edits are always welcome.

File: docs/user-guide/spin.rst (hunk 16 above)
Comment (jgerh) [reply docs/user-guide/spin.rst:11]: LGTM

**Review by terrykong — APPROVED** [review]

**Pull request merged**